docs/spark/tutorials/batch-processing.md

```diff
@@ -0,0 +1,150 @@
+
+1. To use .NET for Apache Spark in an app, install the Microsoft.Spark package. In your console, run the following command:
+
+   ```console
+   dotnet add package Microsoft.Spark
+   ```
+
+## Create a SparkSession
+
+1. Add the following additional `using` statements to the top of the *Program.cs* file in *mySparkBatchApp*:
+
+   ```csharp
+   using System;
+   using Microsoft.Spark.Sql;
+   using static Microsoft.Spark.Sql.Functions;
+   ```
+
+1. Add the following code to establish a new SparkSession. The SparkSession is the entry point to programming Spark with the Dataset and DataFrame API. By calling the `spark` object, you can access Spark and DataFrame functionality throughout your program.
+
+   ```csharp
+   SparkSession spark = SparkSession
+        .Builder()
+        .AppName("GitHub and Spark Batch")
+        .GetOrCreate();
+   ```
+
+## Prepare the data
+
+1. Read the input file into a DataFrame, which is a distributed collection of data organized into named columns. You can set the columns for your data through <xref:Microsoft.Spark.Sql.DataFrame.Schema%2A>. Use <xref:Microsoft.Spark.Sql.DataFrame.Show%2A> to display the data in your DataFrame. Be sure to update the CSV filepath to the location of the GitHub data you downloaded.
+
+   ```csharp
+   DataFrame projectsDf = spark
+       .Read()
+       .Schema("id INT, url STRING, owner_id INT, " +
+       "name STRING, descriptor STRING, language STRING, " +
+       "created_at STRING, forked_from INT, deleted STRING" +
+       "updated_at STRING")
+       .Csv("filepath");
+
+   projectsDf.Show();
+   ```
+
+1. Use <xref:Microsoft.Spark.Sql.DataFrame.Na%2A> to drop rows with NA (null) values, and the <xref:Microsoft.Spark.Sql.DataFrame.Drop%2A> method to remove certain columns from your data. This helps prevent errors if you try to analyze null data or columns that are not relevant to your final analysis.
+
+   ```csharp
+   // Drop any rows with NA values
+   DataFrameNaFunctions dropEmptyProjects = projectsDf.Na();
+   DataFrame cleanedProjects = dropEmptyProjects.Drop("any");
+
+   // Remove unnecessary columns
+   cleanedProjects = cleanedProjects.Drop("id", "url", "owner_id");
+   cleanedProjects.Show();
+   ```
+
+## Analyze the data
+
+1. Spark SQL allows you to make SQL calls on your data. It's common to combine user-defined functions and Spark SQL to apply a user-defined function to all rows of your DataFrame.
+
+   You can specifically call `spark.Sql` to mimic standard SQL calls seen in other types of apps. You can also call methods like <xref:Microsoft.Spark.Sql.DataFrame.GroupBy%2A> and <xref:Microsoft.Spark.Sql.DataFrame.Agg%2A> to specifically combine, filter, and perform calculations on your data.
+
```

Comment:
Seems like it'd be worthwhile to explain what the code below is doing (since it'd be rather long or complex to a new user). Something like the following:

Our goal through this app is to gain some insights about our GitHub projects data. We start off by finding the number of times each language has been forked by grouping our data by language, and then calling the `Avg` method.

We next choose to sort our data in descending order through `Desc()` so that we can more easily see which languages are the most forked.

Next, we want to take a look at how recently the projects have been updated. We register a new user-defined function called _MyUDF_ and compare it with a date _s_referenceDate_ (that we had set earlier in our app) to see if each project was updated before or after that reference date. We use Spark SQL to then call that UDF on each row of our data and to thus analyze each project in our data set.